### PR TITLE
[Impeller] Log non-default graphics backend usages, use IMPORTANT rather than ERROR

### PR DESCRIPTION
**Pull request by dnfield**
Fixes https://github.com/flutter/flutter/issues/142488

- Only logs on iOS if Skia is used instead of Impeller.
- Logs on other platforms if Impeller is used instead of Skia.
- Uses "IMPORTANT" rather than "ERROR" for these logs. This will show up by default since flutter_tools sets ERROR and above as logs to show.
- Adds some tests.
- Makes INFO log print file paths the same as other verbosities.

### CODE REVIEW

File: fml/log_level.h [hunk 1]
```diff
@@ -13,8 +13,11 @@ typedef int LogSeverity;
 constexpr LogSeverity kLogInfo = 0;
 constexpr LogSeverity kLogWarning = 1;
 constexpr LogSeverity kLogError = 2;
-constexpr LogSeverity kLogFatal = 3;
-constexpr LogSeverity kLogNumSeverities = 4;
+// A log that is not an error, is important enough to display even if ordinary
+// info is hidden.
+constexpr LogSeverity kLogImportant = 3;
```
Comment (johnmccutchan) [inline fml/log_level.h:18]: Will this break users who set a log level on the command line? If so, should we add a note in our release notes?

Comment (johnmccutchan) [reply fml/log_level.h:18]: Also,

If thew intent

**Review by johnmccutchan — APPROVED** [review]

**Review by jmagman — COMMENTED** [review]
We should really move the iOS/macOS logging from syslog to os_log.  That's been on my backlog list for a long time...
I *think* `LOG_ALERT` still shows up as an error (`OS_LOG_TYPE_ERROR`)?  When you use IMPORTANT does it still look like:
```
[ERROR:flutter/shell/platform/darwin/graphics/FlutterDarwinContextMetalImpeller.mm] blah
```

File: fml/log_level.h (hunk 1 above)
Comment (dnfield) [reply fml/log_level.h:18]: It shouldn't - I'm replacing the `FATAL` level with `IMPORTANT` - it's not actually possible to ignore FATAL, but I suppose someone who was suppressing `ERROR` will now get `IMPORTANT`. I doubt this is happening much if at all though.

**Comment by dnfield** [comment]
> We should really move the iOS/macOS logging from syslog to os_log. That's been on my backlog list for a long time... I _think_ `LOG_ALERT` still shows up as an error (`OS_LOG_TYPE_ERROR`)? When you use IMPORTANT does it still look like:
> 
> ```
> [ERROR:flutter/shell/platform/darwin/graphics/FlutterDarwinContextMetalImpeller.mm] blah
> ```

The `[ERROR` part is something we're adding to the string getting logged. It will look like `[IMPORTANT...`.

**Comment by dnfield** [comment]
(it's quite possible that Console.app will still show it as some kind of error even though the string won't say ERROR...)

**Review by christopherfujino — APPROVED** [review]
RSLGTM

**Pull request merged**